### PR TITLE
fix: Deleting role members

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Noticed whilst working on the new rbac stuff that deleting of roles is broken. It wasn't caught as there was no test 🤦 
Issue came from the RoleMembership not having an org or team to check for.

Also noticed that Role membership is linked to a user when it should really be linked to an org membership 🤦 🤦 

## Changes

* Adds a test for deleting
* Adds the `organization_member` relation to populate it for new fields (I'll followup later with a migration to remove the old filed) - this way we at least get the cascading delete behaviour
* Fixes the `extract_organization` to consider the filter rewrites of the view in order to find the org.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/permissions.py [hunk 1]
```diff
@@ -19,7 +20,15 @@
...
 
 
-def extract_organization(object: Model) -> Organization:
+def extract_organization(object: Model, view: View) -> Organization:
+    # This is set as part of the TeamAndOrgViewSetMixin to allow models that are not directly related to an organization
+    organization_id_rewrite = getattr(view, "filter_rewrite_rules", {}).get("organization_id")
+    if organization_id_rewrite:
+        for part in organization_id_rewrite.split("__"):
+            if part == "organization_id":
+                continue
```
Comment (Twixes) [inline posthog/permissions.py:29]: In this wouldn't `break` be the fully correct statement?

**Review by Twixes — APPROVED** [review]
Not gonna lie, sounds pretty good to me. I suggest a couple minor improvement, but feel free to merge after addressing

File: posthog/permissions.py (hunk 1 above)
Comment (Twixes) [reply posthog/permissions.py:29]: Or fully fully correctly, we should check if the rewrite ends with `__organization_id`: if it does, remove it; if it doesn't, we shouldn't use the rewrite at all

Comment (benjackwhite) [reply posthog/permissions.py:29]: I guess it doesn't really matter (both work). 

For the other part - if you use the rewrite rule "organization_id": "foo" and it _doesn't_ resolve to an organization ID then its pretty borked anyways? 

Happy to make changes for clarity though

**Pull request merged**